### ISSUE TITLE
`helm dependency build` doesn't work with repository url

### ISSUE THREAD

**Issue by janario** [helm/helm#12920]
I have a helm chart with a dependency that points to a repository URL.

``` 
apiVersion: v2
appVersion: 0.2.0
name: ci-prometheus
version: 0.2.0
type: application
dependencies:
  - name: kube-prometheus-stack
    version: 45.10.1
    repository: "https://prometheus-community.github.io/helm-charts"
    condition: kube-prometheus-stack.prometheus.enabled
```

But at my CI we are running `helm dependency build` to rely on the lock file and not have a surprise changes

https://helm.sh/docs/helm/helm_dependency_build/

The problem is that it forces me to have added it in the repo list:
```
Error: no repository definition for https://prometheus-community.github.io/helm-charts. Please add the missing repos via 'helm repo add' 
```

It should use the URL as in the helm dependency update.

Output of `helm version`:

Output of `kubectl version`:

Cloud Provider/Platform (AKS, GKE, Minikube etc.):

**Comment by SayakMukhopadhyay** [comment]
I came across this issue today and I think this issue is related to the closed issue #7214. There has been a lot of discussion in that issue and it seems that this issue is a regression to that issue.

**Comment by banjoh** [comment]
I added https://github.com/helm/helm/issues/7214#issuecomment-2585797889 comment which clarifies the current state of affairs.

TLDR; automatic cache updates may be removed entirely.

I suggest we close this issue in favour of https://github.com/helm/helm/issues/7214
